### PR TITLE
fix: Fixed the bug that document icon appears as pdf icon

### PR DESCRIPTION
**Pull request by obslee**
as title

Log: Fixed the bug that document icon appears as pdf icon
Bug: https://pms.uniontech.com/bug-view-155717.html

### CODE REVIEW

**Review by itsXuSt — CHANGES_REQUESTED** [review]
缩进异常

**Review by misonlan — DISMISSED** [review]

File: src/dde-file-manager-lib/shutil/fileutils.cpp [hunk 1]
```diff
@@ -2293,6 +2293,22 @@ QList<QStringList> FileUtils::catFstabFileInfo(const QString &grepData)
     return listMountInfo;
 }
 
+QString FileUtils::iconNameFromGio(const QUrl &url)
+{
+    g_autoptr(GFile) gfile = g_file_new_for_uri(url.toString().toLocal8Bit().data());
```
Comment (itsXuSt) [inline src/dde-file-manager-lib/shutil/fileutils.cpp]: url.tostring.**tostdstring**.c_str()

File: src/dde-file-manager-lib/shutil/fileutils.cpp [hunk 2]
```diff
@@ -2293,6 +2293,22 @@ QList<QStringList> FileUtils::catFstabFileInfo(const QString &grepData)
...
+    g_autoptr(GFileInfo) gfileInfo = nullptr;
+    gfileInfo = g_file_query_info(gfile, "standard::icon", G_FILE_QUERY_INFO_NONE, nullptr, nullptr);
```
Comment (itsXuSt) [inline src/dde-file-manager-lib/shutil/fileutils.cpp]: 可合并一行

File: src/dde-file-manager-lib/shutil/fileutils.cpp [hunk 3]
```diff
@@ -2293,6 +2293,22 @@ QList<QStringList> FileUtils::catFstabFileInfo(const QString &grepData)
...
+    if(!gfileInfo)
+        return QString();
+    g_autoptr(GObject) icon = g_file_info_get_attribute_object(gfileInfo, "standard::icon");
+    if(!icon)
+        return QString();
+    auto names = g_themed_icon_get_names(G_THEMED_ICON(icon));
+    if(names && names[0] != nullptr)
```
Comment (itsXuSt) [inline src/dde-file-manager-lib/shutil/fileutils.cpp]: names && names[0] 指针判定无需 /*!= nullptr*/

**Review by itsXuSt — CHANGES_REQUESTED** [review]
编码风格问题较多

File: src/dde-file-manager-lib/shutil/fileutils.cpp (hunk 1 above)
Comment (obslee) [reply src/dde-file-manager-lib/shutil/fileutils.cpp]: done

File: src/dde-file-manager-lib/shutil/fileutils.cpp [hunk 4]
```diff
@@ -2293,6 +2293,21 @@ QList<QStringList> FileUtils::catFstabFileInfo(const QString &grepData)
     return listMountInfo;
 }
 
+QString FileUtils::iconNameFromGio(const QUrl &url)
+{
+    g_autoptr(GFile) gfile = g_file_new_for_uri(url.toString().toStdString().c_str());
```
Comment (Clauszy) [inline src/dde-file-manager-lib/shutil/fileutils.cpp:2298]: url.toString().toStdString().c_str() 危险写法

File: src/dde-file-manager-lib/shutil/fileutils.cpp [hunk 5]
```diff
@@ -2293,6 +2293,21 @@ QList<QStringList> FileUtils::catFstabFileInfo(const QString &grepData)
...
+    g_autoptr(GFileInfo) gfileInfo = g_file_query_info(gfile, "standard::icon", G_FILE_QUERY_INFO_NONE, nullptr, nullptr);
+    if (!gfileInfo)
+        return QString();
+    g_autoptr(GObject) icon = g_file_info_get_attribute_object(gfileInfo, "standard::icon");
+    if (!icon)
+        return QString();
+    auto names = g_themed_icon_get_names(G_THEMED_ICON(icon));
+    if (names)
+        return QString::fromLocal8Bit(names[0]);
```
Comment (Clauszy) [inline src/dde-file-manager-lib/shutil/fileutils.cpp:2307]: 不判断越界？

File: src/dde-file-manager-lib/shutil/fileutils.cpp [hunk 6]
```diff
@@ -2293,6 +2293,21 @@ QList<QStringList> FileUtils::catFstabFileInfo(const QString &grepData)
...
+{
+    g_autoptr(GFile) gfile = g_file_new_for_uri(url.toString().toStdString().c_str());
+    g_autoptr(GFileInfo) gfileInfo = g_file_query_info(gfile, "standard::icon", G_FILE_QUERY_INFO_NONE, nullptr, nullptr);
+    if (!gfileInfo)
+        return QString();
+    g_autoptr(GObject) icon = g_file_info_get_attribute_object(gfileInfo, "standard::icon");
+    if (!icon)
+        return QString();
+    auto names = g_themed_icon_get_names(G_THEMED_ICON(icon));
+    if (names)
```
Comment (misonlan) [inline src/dde-file-manager-lib/shutil/fileutils.cpp:2306]: 这里返回的是char**，需要判断names和names[0]

**Pull request closed without merging**